libtheora/x86 source files getting ignored by .gitignore

### Godot version

4.0

### System information

Ubuntu 22.04

### Issue description

Looks like the /thirdparty/libtheora/x86 directory is accidentally getting ignored by the toplevel .gitignore, specifically as a result of the 'x86/' line below:

```
# Build results
[Dd]ebug/
[Dd]ebugPublic/
[Rr]elease/
[Rr]eleases/
x64/
x86/
[Ww][Ii][Nn]32/
[Aa][Rr][Mm]/
[Aa][Rr][Mm]64/
bld/
[Bb]in/
[Oo]bj/
[Ll]og/
[Ll]ogs/
```

This causes git to miss these source files when adding godot as part of another git project repository, and subsequently the build to fail with the following error:
```
Compiling thirdparty/libtheora/decinfo.c ...
In file included from thirdparty/libtheora/decint.h:22,
                 from thirdparty/libtheora/decinfo.c:21:
thirdparty/libtheora/state.h:52:13: fatal error: x86/x86int.h: No such file or directory
   52 | #   include "x86/x86int.h"
      |             ^~~~~~~~~~~~~~
compilation terminated.
scons: *** [thirdparty/libtheora/decinfo.linuxbsd.editor.x86_64.o] Error 1
scons: building terminated because of errors.
```


### Steps to reproduce

1. Create a new git repo
2. Checkout godot inside the new repo directory
3. git add -u
4. Note that the thirdparty/libtheora/x86 directory was not added

### Minimal reproduction project

n/a

--- a/.gitignore
+++ b/.gitignore
@@ -242,6 +242,8 @@ xcuserdata/
 [Rr]eleases/
 x64/
 x86/
+# Not build results, this is Theora source code.
+!thirdparty/libtheora/x86/
 [Ww][Ii][Nn]32/
 [Aa][Rr][Mm]/
 [Aa][Rr][Mm]64/

